### PR TITLE
Make project dashboard default to enrollments

### PR DESCRIPTION
**Pull request by martha**
## Description

If the user has permission to view enrollments on a project, then navigating to the project should default to bringing up the enrollments page. Otherwise, if they don't have permission, then the current behavior (bringing up the overview page) should be maintained.
https://www.pivotaltracker.com/n/projects/2591838/stories/186877672

### Testing instructions
Here's the list that I used to help me test the different cases- I'll paste it here in case it's helpful to others testing and so you can make sure I haven't missed any:
- [ ] Clicking a project from the projects search
  - [ ] With permission to view enrollments on that project
  - [ ] Without permission to view enrollments on that project
- [ ] Clicking a project from an organization’s list of projects
  - [ ] With permission to view enrollments on that project
  - [ ] Without permission to view enrollments on that project
- [ ] Clicking a project from an enrollment
  - [ ] With permission to view enrollments on that project
  ~Without permission: I believe this cannot happen; if you're already viewing an enrollment on the project then you must have permission to view enrollments on that project~

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [x] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/components/accessWrappers/ProjectRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import Loading from '../elements/Loading';
+import NotFound from '../pages/NotFound';
+
+import { ProjectPermissions } from '@/modules/permissions/types';
+import { useHasProjectPermissions } from '@/modules/permissions/useHasPermissionsHooks';
+import { useProjectDashboardContext } from '@/modules/projects/components/ProjectDashboard';
+import { ensureArray } from '@/utils/arrays';
+import { generateSafePath } from '@/utils/pathEncoding';
+
+/**
```
Comment (martha) [inline src/components/accessWrappers/ProjectRoute.tsx:12]: This file is copied and only lightly modified from `EnrollmentProjectRoute.tsx`. It's also very similar to a few of the other files in that `accessWrappers` directory. Let me know if you would like me to dedicate some time to reducing repeated code here?

File: src/routes/protected.tsx [hunk 2]
```diff
@@ -131,7 +132,17 @@ export const protectedRoutes: RouteNode[] = [
         path: Routes.PROJECT,
         element: <ProjectDashboard />,
         children: [
-          { path: '', element: <Navigate to='overview' replace /> },
+          {
+            path: '',
+            element: (
+              <ProjectRoute
+                permissions={['canViewEnrollmentDetails']}
```
Comment (martha) [inline src/routes/protected.tsx:139]: I had more questions about which is the correct permission to use here. I gathered a list of all the permissions that I thought might be relevant:

- Permissions on the project (`ProjectAccessFieldsFragment`)
  - `canEditEnrollments`
  - `canViewEnrollmentDetails`
  - (No such thing as simply `canViewEnrollments`)
- Other possibly relevant permissions on the HMIS admin dashboard
  - `canViewLimitedEnrollmentDetails` - this does not seem to be attached to a project, but if someone has this permission overall, should they be directed to a project enrollment page, or to the overview?
  - `canViewOpenEnrollmentSummary` which I think is associated with the Root permissions - not associated with a particular project

Comment (gigxz) [reply src/routes/protected.tsx:139]: This is correct, `canViewEnrollmentDetails` is needed to view the Project Enrollments page. `canViewLimitedEnrollmentDetails` would not give someone access to view the full enrollments, it only lets you see limited enrollment details on the enrollment tab on the client dashboard.

**Review by gigxz — APPROVED** [review]

File: src/components/accessWrappers/ProjectRoute.tsx (hunk 1 above)
Comment (gigxz) [reply src/components/accessWrappers/ProjectRoute.tsx:12]: looks good to me

**Pull request merged**